src/main/java/com/onarandombox/MultiverseCore/api/MVWorldManager.java

```diff
@@ -108,11 +109,10 @@ boolean addWorld(String name, Environment env, String seedString, WorldType type
      * @param name The name of the world to remove
      * @param removeFromConfig If true(default), we'll remove the entries from the
      *                         config. If false, they'll stay and the world may come back.
-     * @param deleteWorldFolder If true the world folder will be completely deleted. If false
-     *                          only the contents of the world folder will be deleted
+     * @param keepContents The files you want to keep. (Paper configuration, World directory, or nothing)
      * @return True if success, false if failure.
      */
-    boolean deleteWorld(String name, boolean removeFromConfig, boolean deleteWorldFolder);
+    boolean deleteWorld(String name, boolean removeFromConfig, KeepWorld keepContents);
```

Comment:
Yep definitely should not change the existing interface signature. I suggest marking this as `@deprecated` and creating an overload with the new enum.